ee/tables/execparsers/json/parser.go

```diff
@@ -0,0 +1,47 @@
+//go:build darwin || linux
+// +build darwin linux
+
+package json
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	"golang.org/x/text/encoding/unicode"
+	"golang.org/x/text/transform"
+)
+
+// parseJson implements the parsing logic for JSON data.
+// It follows the same approach as ee/dataflatten/json.go
+func (p *parser) parseJson(reader io.Reader) (any, error) {
+	// Read all data from the reader
+	rawdata, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, fmt.Errorf("reading from reader: %w", err)
+	}
+
+	var data any
+
+	// Check if the raw data is valid JSON
+	if json.Valid(rawdata) {
+		if err := json.Unmarshal(rawdata, &data); err != nil {
+			return nil, fmt.Errorf("unmarshalling json: %w", err)
+		}
+		return data, nil
+	}
+
+	// We don't have valid json data, so try to convert possible utf16 data to utf8
+	rawdata, _, err = transform.Bytes(unicode.UTF16(unicode.LittleEndian, unicode.UseBOM).NewDecoder(), rawdata)
+	if err != nil {
+		return nil, errors.New("invalid json. (Despite attempted transform from utf16 to utf8)")
+	}
+
+	// Try to unmarshal the transformed data
+	if err := json.Unmarshal(rawdata, &data); err != nil {
+		return nil, fmt.Errorf("unmarshalling json: %w", err)
```

Comment:
Nitpick -- maybe worth adding a little bit of extra context to disambiguate between this error and the one on line 30 above --

```suggestion
		return nil, fmt.Errorf("unmarshalling json after transforming from utf16 to utf8: %w", err)
```